contracts/HATVaultsRegistry.sol

```diff
@@ -0,0 +1,396 @@
+        emit SetDefaultChallengeTimeOutPeriod(_defaultChallengeTimeOutPeriod);
+    }
+   
+    /** @notice See {IHATVaultsRegistry-setFeeSetter}. */
+    function setFeeSetter(address _feeSetter) external onlyOwner {
+        feeSetter = _feeSetter;
+        emit SetFeeSetter(_feeSetter);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setWithdrawRequestParams}. */
+    function setWithdrawRequestParams(uint256 _withdrawRequestPendingPeriod, uint256  _withdrawRequestEnablePeriod)
+        external 
+        onlyOwner
+    {
+        if (_withdrawRequestPendingPeriod > 90 days)
+            revert WithdrawRequestPendingPeriodTooLong();
+        if (_withdrawRequestEnablePeriod < 6 hours)
+            revert WithdrawRequestEnabledPeriodTooShort();
+        if (_withdrawRequestEnablePeriod > 100 days)
+            revert WithdrawRequestEnabledPeriodTooLong();
+        generalParameters.withdrawRequestPendingPeriod = _withdrawRequestPendingPeriod;
+        generalParameters.withdrawRequestEnablePeriod = _withdrawRequestEnablePeriod;
+        emit SetWithdrawRequestParams(_withdrawRequestPendingPeriod, _withdrawRequestEnablePeriod);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setClaimFee}. */
+    function setClaimFee(uint256 _fee) external onlyOwner {
+        generalParameters.claimFee = _fee;
+        emit SetClaimFee(_fee);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setWithdrawSafetyPeriod}. */
+    function setWithdrawSafetyPeriod(uint256 _withdrawPeriod, uint256 _safetyPeriod) external onlyOwner { 
+        if (_withdrawPeriod < 1 hours) revert WithdrawPeriodTooShort();
+        if (_safetyPeriod > 6 hours) revert SafetyPeriodTooLong();
+        generalParameters.withdrawPeriod = _withdrawPeriod;
+        generalParameters.safetyPeriod = _safetyPeriod;
+        emit SetWithdrawSafetyPeriod(_withdrawPeriod, _safetyPeriod);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setHatVestingParams}. */
+    function setHatVestingParams(uint256 _duration, uint256 _periods) external onlyOwner {
+        if (_duration >= 180 days) revert HatVestingDurationTooLong();
+        if (_periods == 0) revert HatVestingPeriodsCannotBeZero();
+        if (_duration < _periods) revert HatVestingDurationSmallerThanPeriods();
+        generalParameters.hatVestingDuration = _duration;
+        generalParameters.hatVestingPeriods = _periods;
+        emit SetHatVestingParams(_duration, _periods);
+    }
+
+    /** @notice See {IHATVaultsRegistry-setMaxBountyDelay}. */
+    function setMaxBountyDelay(uint256 _delay) external onlyOwner {
+        if (_delay < 2 days) revert DelayTooShort();
+        generalParameters.setMaxBountyDelay = _delay;
+        emit SetMaxBountyDelay(_delay);
+    }
+
+    /** @notice See {IHATVaultsRegistry-createVault}. */
+    function createVault(
+        IERC20 _asset,
+        address _owner,
+        address _committee,
+        IRewardController _rewardController,
```

Comment:
should we have a default reward controller?

Reply:
why

Reply:
because most vault will probably use the default reward plan, no?